lib/ansible/modules/database/postgresql/postgresql_user.py

```diff
@@ -292,22 +294,14 @@ def user_alter(cursor, module, user, password, role_attr_flags, encrypted, expir
         # Do we actually need to do anything?
         pwchanging = False
         if password is not None:
-            if encrypted == 'ENCRYPTED':
-                if password.startswith('md5'):
-                    if password != current_role_attrs['rolpassword']:
-                        pwchanging = True
-                else:
-                    try:
-                        from passlib.hash import postgres_md5 as pm
-                        if pm.encrypt(password, user) != current_role_attrs['rolpassword']:
-                            pwchanging = True
-                    except ImportError:
-                        # Cannot check if passlib is not installed, so assume password is different
-                        pwchanging = True
-            else:
+            if ((password.startswith('md5') and len(password) == 32+3) or encrypted == 'UNENCRYPTED'):
                 if password != current_role_attrs['rolpassword']:
                     pwchanging = True
 
+            if not pwchanging and encrypted == 'ENCRYPTED':
+                if md5(to_bytes(password) + to_bytes(user)).hexdigest() != current_role_attrs['rolpassword']:
```

Comment:
So, if the password start by 'md5', this trigger a special case (that's the whole point of the PR). but why do we compare the md5 of the password + user with current_role_attrs['rolpassword'], since  current_role_attrs['rolpassword'] will be prefixed with 'md5' ? (ie, this would always change the password)